man/fi_nic.3.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: page
+title: fi_nic(3)
+tagline: Libfabric Programmer's Manual
+---
+{% include JB/setup %}
+
+# NAME
+
+fi_nic \- Fabric network interface card attributes
+
+# NETWORK INTERFACE CARD ATTRIBUTES
+
+The fid_nic structure defines attributes for a struct fi_info that
+is directly associated with underlying networking hardware and may
+be returned directly from calling [`fi_getinfo`(3)](fi_getinfo.3.html).
+The format of fid_nic and the related substructures are defined below.
+
+Note that not all fields of all structures may be available.  Unavailable
+or fields that are not applicable to the indicated device will be set to
+NULL or 0.
+
+```c
+struct fid_nic {
+	struct fid             fid;
+	struct fi_device_attr *device_attr;
+	struct fi_bus_attr    *bus_attr;
+	struct fi_link_attr   *link_attr;
+	void                  *prov_attr;
+};
+
+struct fi_device_attr {
+	char *name;
+	char *device_id;
+	char *device_version;
+	char *vendor_id;
+	char *driver;
+	char *firmware;
+};
+
+struct fi_pci_attr {
+	char *domain_id;
+	char *bus_id;
+	char *device_id;
+	char *function_id;
+};
+
+struct fi_bus_attr {
+	enum fi_bus_type       bus_type;
+	union {
+		struct fi_pci_attr pci;
+	};
+};
+
+struct fi_link_attr {
+	char               *address;
+	size_t             mtu;
+	size_t             speed;
+	enum fi_link_state state;
+	char               *network_type;
+};
+
+```
+
+## Device Attributes
+
+Device attributes are used to identify the specific virtual or hardware
+NIC associated with an fi_info structure.
+
+*name*
+: The operating system name associated with the device.  This may be a
+  logical network interface name (e.g. eth0 or eno1) or an absolute
+  filename.
+
+*device_id*
+: This is a vendor specific identifier for the device or product.
+
+*device_version*
+: Indicates the version of the device_id.
```

Comment:
s/device_id/device/ ?

Reply:
This was referencing the field name, which is device_id.

Reply:
Well, it's the revision of the hardware, not the revision of the device name or ID.